l10n_it_fattura_accompagnatoria/views/report_invoice.xml

```diff
@@ -23,8 +23,10 @@
     <template id="report_fattura_accompagnatoria" inherit_id="account.report_invoice_document">
         <xpath expr="//p[@t-if='o.fiscal_position_id.note']" position="after">
             <p t-if="print_ddt_data">
-                <t t-call="l10n_it_ddt.delivery_data">
-                    <t t-set="ddt" t-value="o"/>
+                <t t-foreach="o.ddt_ids" t-as="current_ddt">
+                    <t t-call="l10n_it_ddt.delivery_data">
+                        <t t-set="ddt" t-value="current_ddt"/>
```

Comment:
Perchè usare i dati dei DDT invece di quelli della fattura?

Reply:
Perché se proviamo a passare i dati della fattura incorriamo in questo errore:
```
Traceback (most recent call last):
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/qweb.py", line 347, in _compiled_fn
    return compiled(self, append, new, options, log)
  File "<template>", line 1, in template_l10n_it_ddt_delivery_data_398
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/ir_qweb.py", line 368, in _get_field
    field = record._fields[field_name]
KeyError: 'weight_manual_uom_id'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/simonev/work/12-instance/odoo/addons/web/controllers/main.py", line 1679, in report_download
    response = self.report_routes(reportname, docids=docids, converter=converter)
  File "/home/simonev/work/12-instance/odoo/odoo/http.py", line 519, in response_wrap
    response = f(*args, **kw)
  File "/home/simonev/work/12-instance/odoo/addons/web/controllers/main.py", line 1620, in report_routes
    pdf = report.with_context(context).render_qweb_pdf(docids, data=data)[0]
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/ir_actions_report.py", line 727, in render_qweb_pdf
    html = self.with_context(context).render_qweb_html(res_ids, data=data)[0]
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/ir_actions_report.py", line 767, in render_qweb_html
    return self.render_template(self.report_name, data), 'html'
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/ir_actions_report.py", line 540, in render_template
    return view_obj.render_template(template, values)
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/ir_ui_view.py", line 1324, in render_template
    return self.browse(self.get_view_id(template)).render(values, engine)
  File "/home/simonev/work/12-instance/odoo/addons/web_editor/models/ir_ui_view.py", line 29, in render
    return super(IrUiView, self).render(values=values, engine=engine, minimal_qcontext=minimal_qcontext)
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/ir_ui_view.py", line 1333, in render
    return self.env[engine].render(self.id, qcontext)
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/ir_qweb.py", line 59, in render
    result = super(IrQWeb, self).render(id_or_xml_id, values=values, **context)
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/qweb.py", line 275, in render
    self.compile(template, options)(self, body.append, values or {})
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/qweb.py", line 349, in _compiled_fn
    raise e
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/qweb.py", line 347, in _compiled_fn
    return compiled(self, append, new, options, log)
  File "<template>", line 1, in template_944_316
  File "<template>", line 2, in body_call_content_315
  File "<template>", line 3, in foreach_314
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/qweb.py", line 349, in _compiled_fn
    raise e
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/qweb.py", line 347, in _compiled_fn
    return compiled(self, append, new, options, log)
  File "<template>", line 1, in template_account_report_invoice_document_365
  File "<template>", line 2, in body_call_content_364
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/qweb.py", line 354, in _compiled_fn
    raise QWebException("Error to render compiling AST", e, path, node and etree.tostring(node[0], encoding='unicode'), name)
odoo.addons.base.models.qweb.QWebException: 'weight_manual_uom_id'
Traceback (most recent call last):
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/qweb.py", line 347, in _compiled_fn
    return compiled(self, append, new, options, log)
  File "<template>", line 1, in template_l10n_it_ddt_delivery_data_398
  File "/home/simonev/work/12-instance/odoo/odoo/addons/base/models/ir_qweb.py", line 368, in _get_field
    field = record._fields[field_name]
KeyError: 'weight_manual_uom_id

Error to render compiling AST
KeyError: 'weight_manual_uom_id'
Template: l10n_it_ddt.delivery_data
Path: /templates/t/table[1]/tr[1]/td[5]/div/span[2]
Node: <span t-field="ddt.weight_manual_uom_id"/>
```

questo si verifica perché il campo `weight_manual_uom_id` non è definito nel model delle fatture `account.invoice` ma in quello dei ddt `stock.picking.package.preparation`.